### ISSUE TITLE
Backup fails when previous backup on same day exists if both occur at same time (in 12h clock)

### ISSUE THREAD

**Issue by ninthspace** [spatie/laravel-backup#290]
This is because the time formats for .zip files created are named as per a 12h clock, not 24h clock. The backup fails either when creating the temporary file, or when moving the temporary file into its final destination.

All occurrences of `Y-m-d-h-i-s` should be replaced with `Y-m-d-H-i-s`, in:

- https://github.com/spatie/laravel-backup/blob/6c9efc48cb63d404b1b9d7bdebfe497217d35fe7/src/Tasks/Backup/TemporaryDirectory.php#L25
- https://github.com/spatie/laravel-backup/blob/6c9efc48cb63d404b1b9d7bdebfe497217d35fe7/tests/Integration/TemporaryDirectoryTest.php#L27
- https://github.com/spatie/laravel-backup/blob/f6a9d8294043212d6b737e434f38acae486a493c/src/Tasks/Backup/BackupJob.php#L172

**Comment by freekmurze** [comment]
Great catch! I never intended to use the 12h clock. Can you PR this?

**Comment by ninthspace** [comment]
Yes :-) This will be my first PR that's not a documentation change.

I've cloned the repo, and `composer install`-ed inside the repo but on running the current tests I get this:

```
> vendor/bin/phpunit
PHP Fatal error:  Cannot make static method PHPUnit_Framework_Assert::assertDirectoryExists() non static in class Spatie\Backup\Test\Integration\TestCase in /home/vagrant/laravel-backup/laravel-backup/tests/Integration/TestCase.php on line 15

Fatal error: Cannot make static method PHPUnit_Framework_Assert::assertDirectoryExists() non static in class Spatie\Backup\Test\Integration\TestCase in /home/vagrant/laravel-backup/laravel-backup/tests/Integration/TestCase.php on line 15
```

Which indicates the three directory assertions in the package's Integration/TestCase.php are trying to override same named methods in PHPUnit_Framework_Assert. Commenting these out makes the tests pass. Am I doing something stupid?

**Comment by freekmurze** [comment]
I had coded up these methods to test if a directory (does not) exist. It seems that in the latest PHPUnit version these asserts were added too. Unlucky coincidence 😄 

I've removed our implementations in `dev-master`. If you pull your repo and runs the tests again all should be well.

**Comment by ninthspace** [comment]
Okay, I have a PR here: https://github.com/spatie/laravel-backup/pull/292

It looks like there are similar issues relating to PHPUnit version in the Travis builds.

**Comment by freekmurze** [comment]
Fix is included in `4.0.4`. I'll fix those tests soon.
